### ISSUE TITLE
graphite tags api: duplicate tag when using telegraf with specific config

### ISSUE THREAD

**Issue by Civil** [VictoriaMetrics/VictoriaMetrics#942]
**Describe the bug**
I was testing graphite API for tags and query to `/tags/autoComplete/tags` sometimes returns duplicate names

**To Reproduce**
1. Start fresh VM 1.49.0 without any extra arguments
2. Start telegraf to collect data (see telegraf config below)
3. Do a request to "http://localhost:8428/tags/autoComplete/tags"

telegraf's config:
```
[global_tags]
  dc = "home" # will tag all metrics with dc=us-east-1
  machine = "pc"
[agent]
  interval = "60s"
  round_interval = true
  metric_batch_size = 1000
  metric_buffer_limit = 10000
  collection_jitter = "0s"
  flush_interval = "10s"
  flush_jitter = "0s"
  precision = ""
  hostname = ""
  omit_hostname = false
[[outputs.influxdb]]
  urls = ["http://127.0.0.1:8428"]
[[outputs.graphite]]
   servers = ["localhost:2003"]
   template = "host.tags.measurement.field"
   graphite_tag_support = true
[[inputs.cpu]]
  percpu = true
  totalcpu = true
  collect_cpu_time = false
  report_active = false
[[inputs.disk]]
  ignore_fs = ["tmpfs", "devtmpfs", "devfs", "iso9660", "overlay", "aufs", "squashfs"]
[[inputs.diskio]]
[[inputs.kernel]]
[[inputs.mem]]
[[inputs.processes]]
[[inputs.swap]]
[[inputs.system]]
```

**Expected behavior**
`["cpu","db","dc","device","fstype","host","machine","mode","name","path"]`

**Actual behavior**
`["cpu","db","dc","device","fstype","host","machine","mode","name","name","path"]`

**Version**
The line returned when passing `--version` command line flag to binary. For example:
```
victoria-metrics-20201206-194153-tags-v1.49.0-0-gae1db8fa
```

**Used command-line flags**
```
flag{name="bigMergeConcurrency", value="0"} 1
flag{name="csvTrimTimestamp", value="1ms"} 1
flag{name="dedup.minScrapeInterval", value="0s"} 1
flag{name="deleteAuthKey", value="secret"} 1
flag{name="denyQueriesOutsideRetention", value="false"} 1
flag{name="dryRun", value="false"} 1
flag{name="enableTCP6", value="false"} 1
flag{name="envflag.enable", value="false"} 1
flag{name="envflag.prefix", value=""} 1
flag{name="finalMergeDelay", value="30s"} 1
flag{name="forceFlushAuthKey", value="secret"} 1
flag{name="forceMergeAuthKey", value="secret"} 1
flag{name="fs.disableMmap", value="false"} 1
flag{name="graphiteListenAddr", value=""} 1
flag{name="graphiteTrimTimestamp", value="1s"} 1
flag{name="http.connTimeout", value="2m0s"} 1
flag{name="http.disableResponseCompression", value="false"} 1
flag{name="http.idleConnTimeout", value="1m0s"} 1
flag{name="http.maxGracefulShutdownDuration", value="7s"} 1
flag{name="http.pathPrefix", value=""} 1
flag{name="http.shutdownDelay", value="0s"} 1
flag{name="httpAuth.password", value="secret"} 1
flag{name="httpAuth.username", value=""} 1
flag{name="httpListenAddr", value=":8428"} 1
flag{name="import.maxLineLen", value="104857600"} 1
flag{name="influx.maxLineSize", value="262144"} 1
flag{name="influxListenAddr", value=""} 1
flag{name="influxMeasurementFieldSeparator", value="_"} 1
flag{name="influxSkipMeasurement", value="false"} 1
flag{name="influxSkipSingleField", value="false"} 1
flag{name="influxTrimTimestamp", value="1ms"} 1
flag{name="insert.maxQueueDuration", value="1m0s"} 1
flag{name="loggerDisableTimestamps", value="false"} 1
flag{name="loggerErrorsPerSecondLimit", value="0"} 1
flag{name="loggerFormat", value="default"} 1
flag{name="loggerLevel", value="INFO"} 1
flag{name="loggerOutput", value="stderr"} 1
flag{name="loggerWarnsPerSecondLimit", value="0"} 1
flag{name="maxConcurrentInserts", value="96"} 1
flag{name="maxInsertRequestSize", value="33554432"} 1
flag{name="maxLabelsPerTimeseries", value="30"} 1
flag{name="memory.allowedBytes", value="0"} 1
flag{name="memory.allowedPercent", value="60"} 1
flag{name="metricsAuthKey", value="secret"} 1
flag{name="opentsdbHTTPListenAddr", value=""} 1
flag{name="opentsdbListenAddr", value=""} 1
flag{name="opentsdbTrimTimestamp", value="1s"} 1
flag{name="opentsdbhttp.maxInsertRequestSize", value="33554432"} 1
flag{name="opentsdbhttpTrimTimestamp", value="1ms"} 1
flag{name="pprofAuthKey", value="secret"} 1
flag{name="precisionBits", value="64"} 1
flag{name="promscrape.config", value=""} 1
flag{name="promscrape.config.dryRun", value="false"} 1
flag{name="promscrape.config.strictParse", value="false"} 1
flag{name="promscrape.configCheckInterval", value="0s"} 1
flag{name="promscrape.consulSDCheckInterval", value="30s"} 1
flag{name="promscrape.disableCompression", value="false"} 1
flag{name="promscrape.disableKeepAlive", value="false"} 1
flag{name="promscrape.discovery.concurrency", value="100"} 1
flag{name="promscrape.discovery.concurrentWaitTime", value="1m0s"} 1
flag{name="promscrape.dnsSDCheckInterval", value="30s"} 1
flag{name="promscrape.dockerswarmSDCheckInterval", value="30s"} 1
flag{name="promscrape.dropOriginalLabels", value="false"} 1
flag{name="promscrape.ec2SDCheckInterval", value="1m0s"} 1
flag{name="promscrape.eurekaSDCheckInterval", value="30s"} 1
flag{name="promscrape.fileSDCheckInterval", value="30s"} 1
flag{name="promscrape.gceSDCheckInterval", value="1m0s"} 1
flag{name="promscrape.kubernetesSDCheckInterval", value="30s"} 1
flag{name="promscrape.maxDroppedTargets", value="1000"} 1
flag{name="promscrape.maxScrapeSize", value="16777216"} 1
flag{name="promscrape.openstackSDCheckInterval", value="30s"} 1
flag{name="promscrape.streamParse", value="false"} 1
flag{name="promscrape.suppressDuplicateScrapeTargetErrors", value="false"} 1
flag{name="promscrape.suppressScrapeErrors", value="false"} 1
flag{name="relabelConfig", value=""} 1
flag{name="retentionPeriod", value="1"} 1
flag{name="search.cacheTimestampOffset", value="5m0s"} 1
flag{name="search.disableCache", value="false"} 1
flag{name="search.latencyOffset", value="30s"} 1
flag{name="search.logSlowQueryDuration", value="5s"} 1
flag{name="search.maxConcurrentRequests", value="16"} 1
flag{name="search.maxExportDuration", value="720h0m0s"} 1
flag{name="search.maxLookback", value="0s"} 1
flag{name="search.maxPointsPerTimeseries", value="30000"} 1
flag{name="search.maxQueryDuration", value="30s"} 1
flag{name="search.maxQueryLen", value="16384"} 1
flag{name="search.maxQueueDuration", value="10s"} 1
flag{name="search.maxStalenessInterval", value="0s"} 1
flag{name="search.maxTagKeys", value="secret"} 1
flag{name="search.maxTagValueSuffixesPerSearch", value="100000"} 1
flag{name="search.maxTagValues", value="100000"} 1
flag{name="search.maxUniqueTimeseries", value="300000"} 1
flag{name="search.minStalenessInterval", value="0s"} 1
flag{name="search.resetCacheAuthKey", value="secret"} 1
flag{name="search.treatDotsAsIsInRegexps", value="false"} 1
flag{name="selfScrapeInstance", value="self"} 1
flag{name="selfScrapeInterval", value="0s"} 1
flag{name="selfScrapeJob", value="victoria-metrics"} 1
flag{name="smallMergeConcurrency", value="0"} 1
flag{name="snapshotAuthKey", value="secret"} 1
flag{name="storageDataPath", value="victoria-metrics-data"} 1
flag{name="tls", value="false"} 1
flag{name="tlsCertFile", value=""} 1
flag{name="tlsKeyFile", value="secret"} 1
flag{name="version", value="false"} 1
```

**Additional context**
PCAP that shows what telegraf sends: https://drive.google.com/file/d/1uHVTVuL_DdAkYRP4k8xjtnGuqJhmM8bH/view?usp=sharing

telegraf versions I've tested (that was a latest master long time ago):
```
❯ /home/civil/go/src/github.com/influxdata/telegraf/telegraf --version
Telegraf unknown (git: master 898487b2)
```

And current master:
```
❯ /home/civil/go/src/github.com/influxdata/telegraf/telegraf --version
Telegraf unknown (git: master 73986acc)
```

This is likely some sort of interoperability issue between telegraf and VM as if I try to remove any section from telegraf's config, duplicate tag disappears.

**Comment by valyala** [comment]
@Civil , this bug should be fixed in the commit 007dbf273d4977fd45252bc8fbf315f88de96131 , which will be included in the next release. In the mean time you can build VictoriaMetrics from this commit according to [these docs](https://victoriametrics.github.io/#how-to-build-from-sources) and verify whether it works as expected.

**Comment by Civil** [comment]
Looks like it's fixed in a current master. Thanks!

**Comment by valyala** [comment]
@Civil , thanks for the confirmation! The bugfix will be included in the next release of VictoriaMetrics.

**Comment by Civil** [comment]
I assume that would be 1.50.0? As I currently put a check in a carbonapi to disable dedupe logic for 1.50.0 or newer.

**Comment by valyala** [comment]
> I assume that would be 1.50.0? As I currently put a check in a carbonapi to disable dedupe logic for 1.50.0 or newer.

Yes.

**Comment by valyala** [comment]
The bugfix has been included in [v1.50.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.
